ghc-toolkit/boot-libs/rts/PrimOps.cmm

```diff
@@ -1441,12 +1441,82 @@ stg_takeMVarzh ( P_ mvar /* :: MVar a */ )
      * and wait until we're woken up.
      */
     if (StgMVar_value(mvar) == stg_END_TSO_QUEUE_closure) {
-        jump stg_raisezh(base_GHCziIOziException_blockedIndefinitelyOnMVar_closure);
+        //if (info == stg_MVAR_CLEAN_info) {
```

Comment:
Why are we commenting out the `dirty_MVAR` logic here? Our gc currently doesn't use clean/dirty info of MutVars, MVars, etc, but dirtying it here won't hurt either. And we can have fewer hacks in this file.